### ISSUE TITLE
Redesign block sync

### ISSUE THREAD

**Issue by arijitAD** [ChainSafe/gossamer#1742]
## Issue summary

- While syncing blocks with connected peers, Node also gets announced block in parallel from multiple peers, which is too far from the current head block. Our present code is processing `announced blocks` and making requests for block data for all announced blocks.

### Problems
- It gets **multiple announced requests for the same block** from multiple peers and is processing all.
- The current Node does not add the `announced blocks response` in the chain because the head is not yet synced to that goal but it still requests those blocks from other peers. It shouldn't make requests for far away blocks.
- This `announced blocks` response is getting added in the list of blocks along with syncing from the current head to be handled which is throwing the error `failed to handle block data : Key not found` while executing blocks because (for announced block data). This creates a block request for the parent block which again causes the same error.

**Example:** 
Node head is at block number 12,800 and the peers are at 8,888,881 block number. The peers will announce 8,888,882 block, Node will accept `announce block` and create a request for blocks after 8,888,881. A parallel requests will be created for blocks after the current head 12,800. In response to the above requests the block's data will be added and processed by processBlockResponse(). After 12,829 it handles block data for 8,888,882 which causes the error "failed to handle block : Key not found" and this error will be handled by handleBlockDataFailure() which creates a request for 8,888,881 block and request for 12,829 by syncAtHead. Due to continuous unwanted announced block syncing as the head is not nearby 8,888,881 the network traffic and CPU usage will increase and slows down the block syncing at the head.

## Other information and links

-

**Comment by noot** [comment]
@arijitAD this should be addressed by #1659

**Comment by noot** [comment]
going to close this for now as a duplicate of #1659 - the refactor will address these issues
